gufe/protocols/protocol.py

```diff
@@ -155,14 +155,14 @@ def _create(
         self,
         stateA: ChemicalSystem,
         stateB: ChemicalSystem,
-        mapping: Optional[Mapping] = None,
+        mapping: Optional[dict[str, ComponentMapping]] = None,
```

Comment:
@dotsdl this is the big change I'm making to Protocol API here.  Previously `Mapping` here wasn't actually well defined/respected.